sonarqube/resource_sonarqube_qualitygate_usergroup_association.go

```diff
@@ -0,0 +1,176 @@
+package sonarqube
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/satori/uuid"
+)
+
+// GetQualityGateUsergroupAssociation for unmarshalling response body from getting quality gate association
+type GetQualityGateUsergroupAssociation struct {
+	Paging Paging                                       `json:"paging"`
+	Groups []GetQualityGateUsergroupAssociationProjects `json:"groups,omitempty"`
+	Users  []GetQualityGateUsergroupAssociationProjects `json:"users,omitempty"`
+}
+
+// GetQualityGateUsergroupAssociationProjects used in GetQualityGateUsergroupAssociation
+type GetQualityGateUsergroupAssociationProjects struct {
+	Login       string `json:"login,omitempty"`
+	Name        string `json:"name"`
+	Description string `json:"description,omitempty"`
+	Selected    bool   `json:"selected"`
+}
+
+// Returns the resource represented by this file.
+func resourceSonarqubeQualityGateUsergroupAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSonarqubeQualityGateUsergroupAssociationCreate,
+		Read:   resourceSonarqubeQualityGateUsergroupAssociationRead,
+		Delete: resourceSonarqubeQualityGateUsergroupAssociationDelete,
+
+		// Define the fields of this schema.
+		Schema: map[string]*schema.Schema{
+			"login_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ExactlyOneOf: []string{"login_name", "group_name"},
+			},
+			"group_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ExactlyOneOf: []string{"login_name", "group_name"},
+			},
+			"gatename": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
I guess the `gatename` should be required
```suggestion
				Required: true,
```